background/get-addon-settings.js

```diff
@@ -654,6 +654,31 @@ chrome.storage.sync.get([...ADDON_SETTINGS_KEYS, "addonsEnabled"], (storageItems
             // Override the preset color if dark comments are not enabled
             addonSettings["comment-color"] = "#FEF49C";
           }
+
+          // Transition v1.38 to v1.39
+          updatePresetIfMatching(
+            settings,
+            4,
+            {
+              "motion-color": "#80B5FF",
+              "looks-color": "#CCB3FF",
+              "sounds-color": "#E19DE1",
+              "events-color": "#FFD966",
+              "control-color": "#FFBE4C",
+              "sensing-color": "#85C4E0",
+              "operators-color": "#7ECE7E",
+              "data-color": "#FFA54C",
+              "data-lists-color": "#FF9966",
+              "custom-color": "#FF99AA",
+              "Pen-color": "#13ECAF",
+              "sa-color": "#34E4D0",
+              "comment-color": "#FEF49C",
+              "input-color": "#202020",
+              text: "colorOnBlack",
+              // "monitors": false
+            },
+            manifest.presets.find((p) => p.id === "black")
+          );
```

Comment:
@mxmou If you want, you can add `//` to more lines. I added it to "monitors" so that the migration happens even if that was enabled by the user but the rest still match. Depending on how terribly blocks will look if this migration does not take place, you may want to make this less strict by commenting out more settings.

Reply:
This migration works the same as the user clicking the Black preset again, so the "monitors" setting will get disabled even if it was enabled prior to the migration.

Reply:
Personally I don't think that users who already have Black applied should have their settings updated to fit the new preset as it's such a drastic change and would make the insertion markers look quite different to how they did before the update. I don't really mind either way though

Reply:
So are you saying that, if `settings["text"]==="colorOnBlack"` at the time of the update, then the new settings (`fillStyle`, `fillOpacity`, `strokeStyle` and `strokeOpacity`) should be set to the Black preset values instead of the setting defaults?

> update the monitor-related settings.

Did you mean "insertion markers" instead of monitor?

Reply:
> So are you saying that, if `settings["text"]==="colorOnBlack"` at the time of the update, then the new settings (`fillStyle`, `fillOpacity`, `strokeStyle` and `strokeOpacity`) should be set to the Black preset values instead of the setting defaults?

Yes, that's what I meant.

Reply:
Okay, when I find some time I will update the setting migration logic and merge this PR.